Assets/MRTK/Providers/WindowsMixedReality/XR2018/Controllers/HPMotionController.cs

```diff
@@ -0,0 +1,309 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using Unity.Profiling;
+using UnityEngine;
+using UnityEngine.XR;
+using System;
+using UnityEngine.XR.WSA.Input;
+
+#if HP_CONTROLLER_ENABLED
+using Microsoft.MixedReality.Input;
+#endif
+
+namespace Microsoft.MixedReality.Toolkit.WindowsMixedReality.Input
+{
+
+#if HP_CONTROLLER_ENABLED
+    internal class MotionControllerState
+    {
+        public MotionControllerState(MotionController mc)
+        {
+            this.MotionController = mc;
+        }
+        public void Update(DateTime currentTime)
+        {
+            this.CurrentReading = MotionController.TryGetReadingAtTime(currentTime);
+        }
+        public MotionController MotionController { get; private set; }
+        public MotionControllerReading CurrentReading { get; private set; }
+    }
+
+    [MixedRealityController(
+        SupportedControllerType.HPMotionController,
+        new[] { Toolkit.Utilities.Handedness.Left, Toolkit.Utilities.Handedness.Right },
+        flags: MixedRealityControllerConfigurationFlags.UseCustomInteractionMappings)]
+    public class HPMotionController : WindowsMixedRealityController
+    {
+        internal MotionControllerState MotionControllerState;
+
+        public HPMotionController(TrackingState trackingState, Toolkit.Utilities.Handedness controllerHandedness, IMixedRealityInputSource inputSource = null, MixedRealityInteractionMapping[] interactions = null)
+            : base(trackingState, controllerHandedness, inputSource, interactions)
+        {
+        }
+
+        /// <inheritdoc />
+        public override MixedRealityInteractionMapping[] DefaultLeftHandedInteractions => new[]
+        {
+            new MixedRealityInteractionMapping(0, "Spatial Pointer", AxisType.SixDof, DeviceInputType.SpatialPointer),
+            new MixedRealityInteractionMapping(1, "Grip Press", AxisType.SingleAxis, DeviceInputType.TriggerPress),
+            new MixedRealityInteractionMapping(2, "Trigger Press", AxisType.Digital, DeviceInputType.Select),
```

Comment:
I think we still want these six (three missing here) mappings from WMRController:

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/498065b67810b5601daf05ba14fd46393bc590ee/Assets/MRTK/Providers/WindowsMixedReality/XR2018/Controllers/WindowsMixedRealityController.cs#L46-L51

plus we can add an actual 0.0 - 1.0 axis for the grip button (in addition to the existing axis which sends a digital button event). Could use DeviceInputType.Grip and DeviceInputType.GripPress to differentiate?

Reply:
Spatial Pointer is mandatory if you want pointers to work for the controller.  Grip pointer is needed if a project is likely to also use the grip pose for grip pointers.

The rest are optional, but I would advice keeping them all for cross-platform purposes.